tests/integration/relation_tests/test_charm_relations.py

```diff
@@ -44,25 +49,54 @@ async def test_deploy_charms(ops_test: OpsTest):
 
     application_charm = await ops_test.build_charm(TEST_APP_CHARM_PATH)
     database_charm = await ops_test.build_charm(".")
-    db_resources = {
-        "mongodb-image": DATABASE_METADATA["resources"]["mongodb-image"]["upstream-source"]
-    }
-    await asyncio.gather(
-        ops_test.model.deploy(
-            application_charm,
-            application_name=APPLICATION_APP_NAME,
-            num_units=2,
-        ),
-        ops_test.model.deploy(
-            database_charm,
-            application_name=DATABASE_APP_NAME,
-            resources=db_resources,
-            num_units=1,
-        ),
-        ops_test.model.deploy(
-            database_charm, application_name=ANOTHER_DATABASE_APP_NAME, resources=db_resources
-        ),
-    )
+
+    required_units = 2
+    app_name = await get_app_name(ops_test)
+    if app_name == ANOTHER_DATABASE_APP_NAME:
+        assert (
+            False
+        ), f"provided MongoDB application, cannot be named {ANOTHER_DATABASE_APP_NAME}, this name is reserved for this test."
+
+    if app_name:
+        db_resources = {
+            "mongodb-image": DATABASE_METADATA["resources"]["mongodb-image"]["upstream-source"]
+        }
+        await asyncio.gather(
+            ops_test.model.deploy(
+                application_charm,
+                application_name=APPLICATION_APP_NAME,
+                num_units=required_units,
```

Comment:
move the duplicated deployments of `APPLICATION_APP_NAME ` and `ANOTHER_DATABASE_APP_NAME ` from this `if/else` chain to the end

Reply:
done, thank you